### PR TITLE
use multiple noarch_platforms

### LINKED ISSUE #1: Installation in Windows fails
### Solution to issue cannot be found in the documentation.

- [X] I checked the documentation.

### Issue

Using conda to install `Pymablock` in Windows results in

```
The following package could not be LibMambaUnsatisfiableError: Encountered problems while solving:

nothing provides __unix needed by pymablock-1.0.0.post0-pyhd8ed1ab_0
Could not solve for environment specs.
The following package could not be installed
└─ pymablock is not installable because it requires
   └─ __unix, which is missing on the system.
```

The [recipe](https://github.com/conda-forge/pymablock-feedstock/blob/main/recipe/meta.yaml#L32) requires `__unix` only in `not win`. However, the installation on Windows ignores this.
Installing in Linux works well.

### Installed packages

```shell
# packages in environment at C:\Users\LocalAdmin\miniconda3\envs\snakes:
#
# Name                    Version                   Build  Channel
brotli                    1.1.0                hcfcfb64_1    conda-forge
brotli-bin                1.1.0                hcfcfb64_1    conda-forge
ca-certificates           2023.08.22           haa95532_0
certifi                   2023.7.22          pyhd8ed1ab_0    conda-forge
colorama                  0.4.6              pyhd8ed1ab_0    conda-forge
contourpy                 1.2.0            py39h1f6ef14_0    conda-forge
cycler                    0.12.1             pyhd8ed1ab_0    conda-forge
exceptiongroup            1.1.3              pyhd8ed1ab_0    conda-forge
fonttools                 4.44.3           py39ha55989b_0    conda-forge
freetype                  2.12.1               hdaf720e_2    conda-forge
importlib-resources       6.1.1              pyhd8ed1ab_0    conda-forge
importlib_resources       6.1.1              pyhd8ed1ab_0    conda-forge
iniconfig                 2.0.0              pyhd8ed1ab_0    conda-forge
intel-openmp              2023.2.0         h57928b3_50497    conda-forge
kiwisolver                1.4.5            py39h1f6ef14_1    conda-forge
kwant                     1.4.4           nompi_py39h53edfc5_101    conda-forge
lcms2                     2.15                 h67d730c_3    conda-forge
lerc                      4.0.0                h63175ca_0    conda-forge
libblas                   3.9.0              19_win64_mkl    conda-forge
libbrotlicommon           1.1.0                hcfcfb64_1    conda-forge
libbrotlidec              1.1.0                hcfcfb64_1    conda-forge
libbrotlienc              1.1.0                hcfcfb64_1    conda-forge
libcblas                  3.9.0              19_win64_mkl    conda-forge
libdeflate                1.19                 hcfcfb64_0    conda-forge
libflang                  5.0.0           h6538335_20180525    conda-forge
libhwloc                  2.9.3           default_haede6df_1009    conda-forge
libiconv                  1.17                 h8ffe710_0    conda-forge
libjpeg-turbo             3.0.0                hcfcfb64_1    conda-forge
liblapack                 3.9.0              19_win64_mkl    conda-forge
libpng                    1.6.39               h19919ed_0    conda-forge
libtiff                   4.6.0                h6e2ebb7_2
```

### Environment info

```shell
active environment : snakes
    active env location : C:\Users\LocalAdmin\miniconda3\envs\snakes
            shell level : 2
       user config file : C:\Users\LocalAdmin\.condarc
 populated config files :
          conda version : 23.10.0
    conda-build version : not installed
         python version : 3.11.5.final.0
       virtual packages : __archspec=1=x86_64
                          __win=0=0
       base environment : C:\Users\LocalAdmin\miniconda3  (writable)
      conda av data dir : C:\Users\LocalAdmin\miniconda3\etc\conda
  conda av metadata url : None
           channel URLs : https://repo.anaconda.com/pkgs/main/win-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/win-64
                          https://repo.anaconda.com/pkgs/r/noarch
                          https://repo.anaconda.com/pkgs/msys2/win-64
                          https://repo.anaconda.com/pkgs/msys2/noarch
          package cache : C:\Users\LocalAdmin\miniconda3\pkgs
                          C:\Users\LocalAdmin\.conda\pkgs
                          C:\Users\LocalAdmin\AppData\Local\conda\conda\pkgs
       envs directories : C:\Users\LocalAdmin\miniconda3\envs
                          C:\Users\LocalAdmin\.conda\envs
                          C:\Users\LocalAdmin\AppData\Local\conda\conda\envs
               platform : win-64
             user-agent : conda/23.10.0 requests/2.31.0 CPython/3.11.5 Windows/10 Windows/10.0.19045 solver/libmamba conda-libmamba-solver/23.11.1 libmambapy/1.5.3
          administrator : False
             netrc file : None
           offline mode : False
```

### PR DESCRIPTION
**Pull request by akhmerov**
closes #1

Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by akhmerov** [comment]
@conda-forge-admin, please rerender

**Comment by akhmerov** [comment]
@isuruf this is following your [suggestion](https://app.gitter.im/#/room/#conda-forge:matrix.org/$17002362755923RJtnE:matrix.org) on gitter. At a glance, the linter seems to contradict the docs that state that `noarch_platforms` is an allowed top-level key. Additionally the linter complains about selectors combined with a noarch package. This was proposed in the initial PR here: https://github.com/conda-forge/staged-recipes/pull/23014#pullrequestreview-1479174545, but doesn't seem to work properly since the package isn't installable on windows.

Is what I'm doing here correct?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -31,6 +31,10 @@ requirements:
     - __win      # [win]
     - __unix     # [not win]
 
+noarch_platforms:
+  - linux_64
+  - win_64
+
```
Comment (isuruf) [inline recipe/meta.yaml]: This should be in `conda-forge.yml`.

**Comment by akhmerov** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml (hunk 1 above)
Comment (akhmerov) [reply recipe/meta.yaml]: My bad! Thanks for the explanation!

**Pull request merged**